website/src/components/edit-word-audio/contributor.tsx

```diff
@@ -163,18 +231,58 @@ export function ContributorEditWordAudio(p: {
             <span>Record audio</span>
           </IconTextButton>
         </div>
-      </div>
+      </form>
 
       {currentTab === "upload" && (
         <UploadAudioSection
           selectedFile={selectedFile}
-          uploadAudio={uploadAudio}
+          uploadAudio={uploadAudioAndReset}
         />
       )}
 
       {currentTab === "record" && (
-        <RecordAudioSection uploadAudio={uploadAudio} />
+        <RecordAudioSection uploadAudio={uploadAudioAndReset} />
+      )}
+
+      {uploadState === "uploading" && (
+        <StatusMessage border="grey">
+          <p>Uploading...</p>
+        </StatusMessage>
+      )}
+
+      {uploadState === "error" && (
+        <StatusMessage border="red">
+          <p>Something went wrong with the upload</p>
+          <button onClick={() => clearUploadError()}>Try again</button>
+        </StatusMessage>
       )}
+    </>
+  )
+}
+
+function StatusMessage({
+  children,
+  border,
+}: {
+  children: ReactNode
+  border: string
+}) {
+  return (
+    <div
+      style={{
+        position: "absolute",
+        top: 0,
+        left: 0,
+        bottom: 0,
+        right: 0,
+        background: "rgba(255,255,255,0.65)",
+        color: "black",
+        padding: 40,
+        backdropFilter: "blur(2px)",
+        border: `4px solid ${border}`,
+      }}
+    >
```

Comment:
This should live in a `...styles.tsx` file

Reply:
Please make this change